src/components/tx/TxData.js

```diff
@@ -67,20 +67,42 @@ class TxData extends React.Component {
         label: 'Funds neighbors',
         ...this.baseItemGraph
       }
-    ]
+    ],
+    ncDeserializer: null,
+    ncLoading: true,
   };
 
   // Array of token uid that was already found to show the symbol
   tokensFound = [];
 
-  componentDidMount = () => {
-    this.calculateTokens();
+  componentDidMount = async () => {
+    await this.handleDataFetch();
   }
 
-  componentDidUpdate = (prevProps) => {
+  componentDidUpdate = async (prevProps) => {
     if (prevProps.transaction !== this.props.transaction) {
-      this.calculateTokens();
+      await this.handleDataFetch();
+    }
+  }
+
+  handleDataFetch = async () => {
+    this.calculateTokens();
+    await this.handleNanoContractFetch();
+  }
+
+  handleNanoContractFetch = async () => {
+    this.setState({ ncLoading: true });
+    if (this.props.transaction.version !== hathorLib.constants.NANO_CONTRACTS_VERSION) {
+      this.setState({ ncLoading: false });
+      return;
     }
```

Comment:
There is a possibility of the "nano contract data loading" message flashing for transactions that are not nano contracts, maybe we should use this.

```suggestion
    if (this.props.transaction.version !== hathorLib.constants.NANO_CONTRACTS_VERSION) {
      this.setState({ ncLoading: false });
      return;
    }
    this.setState({ ncLoading: true });
```

Reply:
Makes sense: https://github.com/HathorNetwork/hathor-explorer/pull/253/commits/8bdd32b7091d3cc97b22a5564cb6bda00b835565